File: placementprepration/tree/daimeter.cpp
#include<iostream>
using namespace std;
struct Node{
    int data;
    struct Node*right;
    struct Node*left;
    Node(int val){
        this->data=val;
        this->left=NULL;
        this->right=NULL;
    }
};
Node*BuildTree(Node*root){
    int data;
    cout<<"Enetr the data "<<endl;
    cin>>data;
    root=new Node(data);
    if(root==NULL){
        return ;
    }
    cout<<"Enter the data in left of "<<data<<endl;
    root->left=BuildTree(root->left);
    cout<<"enetr the data in right of "<<data<<endl;
    root->right=BuildTree(root->right);
    return root;
}
int daimeter(Node*root){
    
}
int main(){
    Node*root=NULL;
    root=BuildTree(root);


    return 0;

}

File: placementprepration/tree/daimeter.py
class Node:
    def __init__(self, val):
        self.data = val
        self.left = None
        self.right = None


def diameter(root):
    max_dia = 0

    def height(node):
        nonlocal max_dia
        if not node:
            return 0
        lh = height(node.left)
        rh = height(node.right)
        max_dia = max(max_dia, lh + rh + 1)
        return max(lh, rh) + 1

    height(root)
    return max_dia


if __name__ == "__main__":
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    print(diameter(root))
